app/routes/surveys/AddSubmissionRoute.js

```diff
@@ -27,9 +43,7 @@ const mapDispatchToProps = { submitFunction: addSubmission };
 
 export default compose(
   replaceUnlessLoggedIn(LoginPage),
-  prepare(({ params: { surveyId } }, dispatch) => dispatch(fetch(surveyId)), [
-    'params.surveyId'
-  ]),
+  prepare(loadData, ['params.surveyId', 'currentUser.id']),
```

Comment:
Add `fetching` like we discussed in the other PR

Reply:
:+1:

Reply:
Does adding 'fetching' work out of the box here? Other places I've sent `state.myApp.fetching` down as a prop and returned a `<LoadingIndicator />` if it was true.

Reply:
Something like this?

https://github.com/webkom/lego-webapp/blob/master/app/routes/surveys/SurveyRoute.js